templates/cluster_inventory.j2

```diff
@@ -12,7 +12,7 @@ localhost ansible_connection=local ansible_python_interpreter=python
 {% for group_data in output.output_value %}
 [{{ cluster_stack.stack.stack_name}}_{{ group_data.group }}]
 {% for node_data in group_data.nodes %}
-{{node_data.name}} ansible_host={{ node_data.ip }}
+{{node_data.name}} ansible_host={{ node_data.ip }} ansible_ssh_user={{ cluster_groups | selectattr("name", "equalto", group_data.group) | map(attribute='user') | join }}
 {% endfor %}
 
```

Comment:
To support a default it's going to get a bit more complicated. How about using `{% set user=... %}` to break it down a bit?

Alternatively, set group vars directly, looping over cluster_groups:

```
[<group>:vars]
ansible_user=<user>
```